### PR TITLE
 [WFLY-6629]: Create test for [WFCORE-1479]: Jar sub deployment has as first dependency parent ear

### PR DESCRIPTION
**Pull request by vpakan**
JIRA: https://issues.jboss.org/browse/WFLY-6629

Depends on https://github.com/wildfly/wildfly-core/pull/1505

Contains test for [WFCORE-1479] issue

### CODE REVIEW

**Comment by vpakan** [comment]
Test failures  are expected since this commit depends on https://github.com/wildfly/wildfly-core/pull/1505

**Comment by ctomc** [comment]
this is ok to test

**Comment by ctomc** [comment]
this ok to test

**Comment by ctomc** [comment]
this is ok to test

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/deployment/classloading/ear/subdeployments/SubDeploymentAvailableInClassPathTestCase.java [hunk 1]
```diff
@@ -134,6 +137,28 @@ public static EnterpriseArchive createEarWithMultipleWars() {
...
+        // TODO: Currently, due to an issue in AS7 integration with Arquillian, the web-app context and the
+        // .ear name should be the same or else you run into test deployment failures like:
+        // Caused by: java.lang.IllegalStateException: Error launching test at
+        // http://127.0.0.1:8080/<earname>/ArquillianServletRunner?outputMode=serializedObject&className=org.jboss.as.test.spec.ear.classpath.unit.SubDeploymentAvailableInClassPathTestCase&methodName=testEjbClassAvailableInServlet. Kept on getting 404s.
+        // @see https://issues.jboss.org/browse/AS7-367
+        final EnterpriseArchive ear = ShrinkWrap.create(EnterpriseArchive.class, WEB_APP_CONTEXT_THREE + ".ear");
+        ear.addAsModule(servletLogicJar);
+        ear.addAsModule(war);
+        ear.addAsLibrary("javax/naming/javax-naming-test-impl.jar", "javax-naming-test-impl.jar");
+        logger.info(ear.toString(true));
```
Comment (ctomc) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/deployment/classloading/ear/subdeployments/SubDeploymentAvailableInClassPathTestCase.java]: please remove this as it only adds noise to testsuite.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/deployment/classloading/ear/subdeployments/servlet/EjbInvokeToStringServlet.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+/**
+ * User: jpai
```
Comment (ctomc) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/deployment/classloading/ear/subdeployments/servlet/EjbInvokeToStringServlet.java]: I doubt Jaikiran wrote this :)

**Comment by vpakan** [comment]
Comments fxed

**Comment by simkam** [comment]
retest this please

**Comment by istudens** [comment]
There is a binary jar file added to this PR, this looks weird.

**Comment by istudens** [comment]
It probably needs to be done by Shrinkwrap.

**Comment by simkam** [comment]
this add binary jar and fails due to checkstyle errors: superseded by #9329 
can someone close this PR? I can't do it

**Pull request closed without merging**